datafusion/core/src/physical_optimizer/pruning.rs

```diff
@@ -487,6 +492,7 @@ fn rewrite_expr_to_prunable(
     column_expr: &Expr,
     op: Operator,
     scalar_expr: &Expr,
+    schema: DFSchema,
```

Comment:
Could you please update the list of rules in the docstring of this function to include the new logic for casts?